cmake/SFMLConfig.cmake.in

```diff
@@ -0,0 +1,148 @@
+# If nothing is specified, the version won't be checked (i.e. any version will be accepted).
+# example:
+#   find_package(SFML COMPONENTS ...)     # no specific version required
+#   find_package(SFML 2 COMPONENTS ...)   # any 2.x version
+#   find_package(SFML 2.4 COMPONENTS ...) # version 2.4 or greater
+#
+# By default, the dynamic libraries of SFML will be found. To find the static ones instead,
+# you must set the SFML_STATIC_LIBRARIES variable to TRUE before calling find_package(SFML ...).
+# You don't need to deal with SFML's dependencies when linking your targets against SFML libraries,
+# they will all be configured automatically, even if you use SFML static libraries.
+# example:
+#   set(SFML_STATIC_LIBRARIES TRUE)
+#   find_package(SFML 2 COMPONENTS network system)
+#
+# On macOS by default CMake will search for frameworks. If you want to use static libraries and have installed
+# both SFML frameworks and SFML static libraries, your must set CMAKE_FIND_FRAMEWORK to "NEVER" or "LAST"
+# in addition to setting SFML_STATIC_LIBRARIES to TRUE. Otherwise CMake will check the frameworks bundle config and
+# fail after finding out that it does not provide static libraries. Please refer to CMake documentation for more details.
+#
+# Additionally, keep in mind that SFML frameworks are only available as release libraries unlike dylibs which
+# are available for both release and debug modes.
+#
+# If SFML is not installed in a standard path, you can use the SFML_DIR CMake variable
+# to tell CMake where SFML's config file is located (PREFIX/lib/cmake/SFML for a library-based installation,
+# and PREFIX/SFML.framework/Resources/CMake on macOS for a framework-based installation).
+#
+# Output
+# ------
+#
+# This script defines the following variables:
+# - For each specified module XXX (system, window, graphics, network, audio, main):
+#   - SFML_XXX_FOUND:  true if either the debug or release library of the xxx module is found
+# - SFML_FOUND:        true if all the required modules are found
+#
+# And the following targets:
+# - For each specified module XXX (system, window, graphics, network, audio, main):
+#   - sfml-XXX
+# The SFML targets are the same for both Debug and Release build configurations and will automatically provide
+# correct settings based on your currently active build configuration. The SFML targets name also do not change
+# when using dynamic or static SFML libraries.
+#
+# When linking against a SFML target, you do not need to specify indirect dependencies. For example, linking
+# against sfml-graphics will also automatically link against sfml-window and sfml-system.
+#
+# example:
+#   find_package(SFML 2 COMPONENTS graphics audio REQUIRED)
+#   add_executable(myapp ...)
+#   target_link_libraries(myapp sfml-graphics sfml-audio)
+
+if (NOT SFML_FIND_COMPONENTS)
+    message(FATAL_ERROR "find_package(SFML) called with no component")
+endif()
+
+set(FIND_SFML_PATHS
+    "${CMAKE_CURRENT_LIST_DIR}/../.."
```

Comment:
I'm not sure what this is for? as I understand it CMAKE_CURRENT_LIST_DIR is the location of the CMakeLists.txt currently being processed. Is this perhaps a user-specific path?

Reply:
When the user calls find_package() and this file is being used, CMAKE_CURRENT_LIST_DIR is actually the directory containing this config file. Which is PREFIX/lib/CMake/SFML, so the ../.. allows searching dependencies in PREFIX/lib. 

This makes me think that if the user chooses to install dependencies in another directory though (SFML_DEPENDENCIES_INSTALL_PREFIX option) this would not work. We could consider that SFML_DEPENDENCIES_INSTALL_PREFIX must also be in FIND_SFML_PATHS. Either as relative or absolute path, or both, I don’t know yet...